test/fixtures/amp-bind-integrations.html

```diff
@@ -37,7 +37,7 @@
 
   <p>AMP-IMG TEST</p>
   <button on="tap:AMP.setState(imageSrc='http://www.google.com/image2')" id="changeImgSrcButton"></button>
-  <button on="tap:AMP.setState(imageSrc='__amp_source_origin')" id="invalidSrcButton"></button>
+  <button on="tap:AMP.setState(imageSrc='?__amp_source_origin')" id="invalidSrcButton"></button>
```

Comment:
Turns out the point of the `__amp_source_origin` check is to prevent it from being used as a query param, check out `url.js#checkCorsUrl`.